src/main/java/com/google/sps/DataGraph.java

```diff
@@ -352,50 +403,57 @@ private void dfsVisit(GraphNode gn, Map<GraphNode, Boolean> visited, int depthRe
-   * of the graph. If the radius is less than 0 or the node specified isn't present, return an empty
-   * graph. Since maxDepth was implemented as DFS, we use BFS for *diversity*. Here, we only
-   * consider children of children and parents of parents.
+   * Returns a MutableGraph from given nodes that are at most a certain radius from a given nodes.
+   * If the radius is less than 0 or the nodes specified aren't present, return an empty graph.
+   *
+   * <p>Here, we only consider children of children and parents of parents. If a node doesn't exist
+   * in the graph, we skip it. If all the nodes in the collection don't exist in the graph, we
+   * return an empty graph.
    *
-   * @param name the name of the node to search for
+   * @param names the names of the nodes whose descendants within radius distance and all associated
+   *     edges should be included in the graph
    * @param radius the distance from the node to search for parents and children
    * @return a graph comprised of only nodes and edges within a certain distance from the specified
-   *     node (or roots if name is ""). Empty if radius is less than 0 or if the node isn't found.
+   *     node. Empty if radius is less than 0 or if the node isn't found. Returns a graph with a
+   *     depth of at most the radius when names is empty.
    */
-  public MutableGraph<GraphNode> getReachableNodes(String name, int radius) {
-    if (radius < 0) {
+  public MutableGraph<GraphNode> getReachableNodes(Collection<String> names, int radius) {
+    if (radius < 0 || names == null) {
       return GraphBuilder.directed().build(); // If max depth below 0, then return an emtpy graph
     }
 
     HashMap<String, GraphNode> graphNodesMap = this.graphNodesMap();
-    if (name.length() != 0 && !graphNodesMap.containsKey(name)) {
-      return GraphBuilder.directed()
-          .build(); // If the specified node is not found, return an empty graph
-    }
-
-    // HashSet should have expected O(1) lookup, changed from HashMap for space
-    // Two different sets are needed because of cases where a node is both a parent
-    // and a child
-    HashSet<GraphNode> visitedChildren = new HashSet<>();
-    HashSet<GraphNode> visitedParents = new HashSet<>();
-
     HashSet<GraphNode> nextLayerChildren = new HashSet<GraphNode>();
     HashSet<GraphNode> nextLayerParents = new HashSet<GraphNode>();
 
-    MutableGraph<GraphNode> graph = this.graph();
-    if (name.length() == 0) {
+    if (names.size() == 0) {
       List<GraphNode> rootNodes =
           this.roots().stream()
               .map(rootName -> this.graphNodesMap().get(rootName))
               .collect(Collectors.toList());
       nextLayerChildren.addAll(rootNodes);
-    } else {
-      GraphNode tgtNode = graphNodesMap.get(name);
-      nextLayerChildren.add(tgtNode);
-      nextLayerParents.add(tgtNode);
     }
 
+    for (String name : names) {
```

Comment:
Optional but you could also use Collections.stream here, with an additional filter, right?